### PR TITLE
Migrate v1beta1 RunResult to Unversioned Package

### PR DESCRIPTION
**Pull request by JeromeJu**
This commit migrates the v1beta1 RunResult to the unversioned result
package since it is no longer a struct in api used for v1beta1 resource types.
The RunResult struct was previously PipelineResourceResult and it has been renamed
and  moved to an unversioned package because it is no longer used in apis.

The old struct `PipelineResourceResult` is kept in v1beta1 for backward compatibility.

follows up: https://github.com/tektoncd/pipeline/pull/6434
fixes: https://github.com/tektoncd/pipeline/issues/6197
/kind misc

# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [n/a] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [n/a] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Review by lbernick — APPROVED** [review]
thanks Jerome, can you please add a bit more detail to the commit message/pr description on the reason for this change?

File: pkg/apis/pipeline/v1beta1/taskrun_types.go [hunk 1]
```diff
@@ -255,7 +255,7 @@ type TaskRunStatusFields struct {
 	// Deprecated: this field is not populated and is preserved only for backwards compatibility
 	// +optional
 	// +listType=atomic
-	ResourcesResult []RunResult `json:"resourcesResult,omitempty"`
+	ResourcesResult []PipelineResourceResult `json:"resourcesResult,omitempty"`
```
Comment (jerop) [inline pkg/apis/pipeline/v1beta1/taskrun_types.go:258]: why did you change back to `PipelineResourceResult`?

Comment (JeromeJu) [reply pkg/apis/pipeline/v1beta1/taskrun_types.go:258]: I thought that using `PipelineResourceResult` could keep the previous reference for `ResourcesResult` which uses `PipelineResourcesResult` and we wouldn't need to import the newly created result package here. But if it made it more confusing I would be happy to change it back.

**Comment by JeromeJu** [comment]
> 

Thanks Billy. I thought for this we were aliasing the old struct with PipelineResourceResult in https://github.com/tektoncd/pipeline/blob/cb9976818151b49b2c8303f91b4f7a7821ec4960/pkg/apis/pipeline/v1beta1/resource_types.go#L43

> Is there any down side to just freezing the fields in place as deprecated in the current package? (we could also move the deprecated types to a separate file in the same package if you're looking to separate out deprecated fields).

And for this since we are going to swap the storage version as this struct is used in [pkg/pod/status](https://github.com/tektoncd/pipeline/blob/cb9976818151b49b2c8303f91b4f7a7821ec4960/pkg/pod/status.go#L265,) we would need to either migrate this to v1 or an unversioned package moving forward 😿 

Does it make sense that with the aliased `v1beta1.PipelineResourceResult` left in v1beta1 for backward compatibility while moving the renamed `RunResult` struct to an unversioned package for keeping the apis independent yet backward compatible? Happy to move it to a more proper place if there is under the apis 🙏

**Comment by wlynch** [comment]
> And for this since we are going to swap the storage version as this struct is used in [pkg/pod/status](https://github.com/tektoncd/pipeline/blob/cb9976818151b49b2c8303f91b4f7a7821ec4960/pkg/pod/status.go#L265,) we would need to either migrate this to v1 or an unversioned package moving forward 😿

If it's used in the v1 API, package versioned or not it's still effectively a part of the v1 API. 😥 I'd be careful with how these are referenced across v1. 

> Does it make sense that with the aliased `v1beta1.PipelineResourceResult` left in v1beta1 for backward compatibility while moving the renamed `RunResult` struct to an unversioned package for keeping the apis independent yet backward compatible? Happy to move it to a more proper place if there is under the apis 🙏

Makes sense to me!

You'd need type aliases for each type/field moved to the new package. Otherwise `PipelineResourceResult` would resolve, but underlying types like `ResultType`, `RunResult`, etc. won't.

File: pkg/apis/pipeline/v1beta1/resource_types.go [hunk 2]
```diff
@@ -17,71 +17,18 @@ limitations under the License.
...
 	resource "github.com/tektoncd/pipeline/pkg/apis/resource/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/result"
 	v1 "k8s.io/api/core/v1"
 )
 
-// RunResult is used to write key/value pairs to TaskRun pod termination messages.
-// The key/value pairs may come from the entrypoint binary, or represent a TaskRunResult.
-// If they represent a TaskRunResult, the key is the name of the result and the value is the
-// JSON-serialized value of the result.
-type RunResult struct {
```
Comment (wlynch) [inline pkg/apis/pipeline/v1beta1/resource_types.go:32]: Also need an alias for RunResult for completeness. (ideally you should be able to make this change without updating any of its usage if you wanted).

**Review by wlynch — CHANGES_REQUESTED** [review]
One small thing, otherwise lgtm.

File: pkg/apis/pipeline/v1beta1/resource_types.go (hunk 2 above)
Comment (JeromeJu) [reply pkg/apis/pipeline/v1beta1/resource_types.go:32]: Makes sense, thanks. I just restored that and aliased.

**Review by wlynch — COMMENTED** [review]
/lgtm
